src/App.js

```diff
@@ -13,37 +12,16 @@ import { NotificationPage } from './pages/notification';
 
 // Components
 import { Navbar } from './components/Navbar';
-import AuthRoute from './util/AuthRoute';
-
-// Redux
-import store from './redux/store';
-import { SET_AUTHENTICATED } from './redux/types';
-import {
-  logoutUserAction,
-  getUserDataAction,
-} from './redux/actions/userActions';
+import { AuthRoute } from './util/AuthRoute';
 
 // Context
 import { ContextProvider } from './components/context/ContextProvider';
 
 axios.defaults.baseURL = '/';
-const token = localStorage.Token;
-const handle = localStorage.Handle;
-if (token && handle) {
-  const decodedToken = jwtDecode(token);
-  if (decodedToken.exp * 1000 < Date.now()) {
-    store.dispatch(logoutUserAction());
-    window.location.href = '/login';
-  } else {
-    store.dispatch({ type: SET_AUTHENTICATED });
-    axios.defaults.headers.common['Authorization'] = token;
-    store.dispatch(getUserDataAction(handle));
-  }
-}
 
 export const App = () => (
-  <ContextProvider>
-    <Router>
+  <Router>
+    <ContextProvider>
```

Comment:
Router is technically a Provider. Could be moved onboard ContextProvider

Reply:
Fixed